base/access_provider/sync_to_target/naming_hint/generator.go

```diff
@@ -20,17 +20,32 @@ type UniqueGenerator interface {
 	Generate(ap *sync_to_target.AccessProvider) (map[string]string, error)
 }
 
+// uniqueGenerator implement a Generate method that generates unique names that can be used to create access elements
+// If defined a prefix is used when generating names
+// If the naming hint of an accessProvider is specified the naming hint will be reformed to a name only consisting valid characters
+// After that a validation is executed to check if the valid name should be post fixed with a unique ID.
+// The postfixes start with two splitCharacters and end with a 4 character hexadecimal number. Note that this is the only place that 2 splitCharacters can be used after each other
+// If the accessProviders are provided in ascending order by ID. The algorithm will check if an already existing postfix exist in the actual name. If that is the case the currentpost fix will be used.
+// Later created access providers will end up with a higher post fix if they are reusing the same valid name.
+// Example:
```

Comment:
Fixing typos etc:

// uniqueGenerator implements a Generate method which generates unique names that can be used to create access elements.
// If defined, a prefix is used when generating the names.
// If the naming hint of an accessProvider is specified, the naming hint will be reformed to a name only consisting valid characters.
// After that, a validation is executed to check if the valid name should be post-fixed with a unique ID.
// The post-fixes start with two splitCharacters and end with a 4 character hexadecimal number. Note that this is the only place that 2 splitCharacters can be used after each other.
// The algorithm has the prerequisite that accessProviders are provided in ascending order by ID.
// The algorithm will check if an already existing postfix exists in the actual name. If that is the case, the current post-fix will be used.
// Later created access providers will end up with a higher post-fix if they are reusing the same valid name.
// Example: